test/common/utxobased/db/NewProcessor.spec.ts

```diff
@@ -0,0 +1,209 @@
+
+    processor = await makeNewProcessor({ disklet })
+    assertNumAddresses(0)
+    await assertLastUsedByFormatPathEmpty()
+  })
+
+  it('insert address to baselets', async () => {
+    const storage = {}
+    const disklet = makeMemoryDisklet(storage)
+    processor = await makeNewProcessor({ disklet })
+
+    // insert an unused address without a path
+    const address1: IAddress = {
+      scriptPubkey: 'testscriptpubkey1',
+      networkQueryVal: 0,
+      lastQuery: 0,
+      lastTouched: 0,
+      used: false,
+      balance: ''
+    }
+    await processor.insertAddress(address1)
+    assertNumAddresses(0)
+    await assertLastUsedByFormatPathEmpty()
+    const [processorAddress1] = await processor.fetchAddresses({
+      scriptPubkey: 'testscriptpubkey1'
+    })
+    expect(processorAddress1.scriptPubkey).to.eqls(address1.scriptPubkey)
+
+    // insert an unused address with a path
+    const address2: IAddress = {
+      scriptPubkey: 'testscriptpubkey2',
+      networkQueryVal: 0,
+      lastQuery: 0,
+      path: { format: 'bip44', changeIndex: 0, addressIndex: 0 },
+      lastTouched: 0,
+      used: false,
+      balance: ''
+    }
+    await processor.insertAddress(address2)
+    assertNumAddresses(1)
+    await assertLastUsedByFormatPathEmpty()
+    const [processorAddress2] = await processor.fetchAddresses({
+      scriptPubkey: 'testscriptpubkey2'
+    })
+    expect(processorAddress2.scriptPubkey).to.eqls(address2.scriptPubkey)
+
+    // insert a used address with a conflicting path
+    const address3: IAddress = {
+      scriptPubkey: 'testscriptpubkey3',
+      networkQueryVal: 0,
+      lastQuery: 0,
+      path: { format: 'bip44', changeIndex: 0, addressIndex: 0 },
+      lastTouched: 0,
+      used: true,
+      balance: ''
+    }
+    let threw = false
+    try {
+      await processor.insertAddress(address3)
+    } catch (err) {
+      expect(err.message).to.equal(
+        'Attempted to save address with an existing path, but different script pubkey'
+      )
+      threw = true
+    }
+    expect(threw).to.be.true
+    assertNumAddresses(1)
+    const [processorAddress3] = await processor.fetchAddresses({
+      scriptPubkey: 'testscriptpubkey3'
+    })
+    expect(processorAddress3).to.be.undefined
+
+    // insert a used address with a path
+    const address4: IAddress = {
+      scriptPubkey: 'testscriptpubkey3',
+      networkQueryVal: 0,
+      lastQuery: 0,
+      path: { format: 'bip44', changeIndex: 0, addressIndex: 1 },
+      lastTouched: 0,
+      used: true,
+      balance: ''
+    }
+    await processor.insertAddress(address4)
+    assertNumAddresses(2)
+    const [processorAddress4] = await processor.fetchAddresses({
+      scriptPubkey: 'testscriptpubkey3'
+    })
+    expect(processorAddress4.scriptPubkey).to.eqls(address4.scriptPubkey)
+    const lastUsed = await processor.lastUsedIndexByFormatPath({
+      format: 'bip44',
+      changeIndex: 0
+    })
+    expect(lastUsed).to.eql(1)
```

Comment:
Why not refactor your `assertLastUsedByFormatPathEmpty(): void` into a `assertLastUsedByFormatPath(toBe: number | undefined): void` and use it for this check as well?

Reply:
Done.